File: user-manager/user-manager-rest-api/test/TestConfigManager.ts
/**
 * @project:   Butterfly
 * @author:    DStack Group
 * @module:    user-manager-rest-api
 * @fileName:  TestConfigManager.ts
 * @created:   2019-03-07
 *
 * --------------------------------------------------------------------------------------------
 * Copyright (c) 2019 DStack Group.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 * --------------------------------------------------------------------------------------------
 *
 * @description:
 */

import { AbstractConfigManager } from '../src/config';

export class TestConfigManager extends AbstractConfigManager {
  private configMap = new Map<string, string>();

  protected readConfigValue(property: string): string | undefined {
    return this.configMap.get(property);
  }

  setProperty(property: string, value: string) {
    this.configMap.set(property, value);
  }
}

File: butterfly/gitlab-producer/src/test/java/it/unipd/dstack/butterfly/producer/gitlab/testutils/TestConfigManager.java
package it.unipd.dstack.butterfly.producer.gitlab.testutils;

import it.unipd.dstack.butterfly.config.AbstractConfigManager;

import java.util.HashMap;
import java.util.Map;

public class TestConfigManager extends AbstractConfigManager {
    private Map<String, String> configMap = new HashMap<>();
    /**
     * Reads a value from a configuration resource.
     *
     * @param property
     * @return the configuration value corresponding to the given property.
     */
    @Override
    protected String readConfigValue(String property) {
        return this.configMap.get(property);
    }

    public void setProperty(String property, String value) {
        this.configMap.put(property, value);
    }
}
